### DISCUSSION TITLE
Obtaining ND array pyramids using Xarray and Zarr

### DISCUSSION THREAD

**Discussion by jesdial-tech** [carbonplan/ndpyramid#144] [General]
Hi all
Mi name is Jesús de Diego and I work for the Spanish company Technosylva. We are creating daily forecast in NetCDF format and are looking for an alternative way for publishing that info. to be consumed by our web applications. I have to say that I do not have previous experience with NetCDF nor zarr formats.
I found out this really impressive application https://maps.demo.carbonplan.org/ and want to test the ndpyramids arrays using our NetCDF. I was able to create a zarr file using a set of netCDF files containing the same set of variables for a different time steps.
<img width="607" alt="image" src="https://github.com/user-attachments/assets/400f78ab-6af2-4ee8-8a4f-ffef4d1f3161">

However, when I try to create the nd pyramids, regardles the different method I use, I get an error:

- When using pyramid resample,:
`output = r'C:/Temporal/zarr'
levels = 10
resampled_pyramid = pyramid_resample(ds, x="longitude", y="latitude", levels=levels, resampling="nearest")
resampled_pyramid.to_zarr(output, consolidated=True, mode='w')`
I get:
`NotImplementedError: 'item' is not yet a valid method on dask arrays`

-When using pyramid_reproject:
`output = r'C:/Temporal/zarr'
levels = 10
pyramid = pyramid_reproject(ds, levels=2)
#resampled_pyramid = pyramid_resample(ds, x="longitude", y="latitude", levels=levels, resampling="nearest")
resampled_pyramid.to_zarr(output, consolidated=True, mode='w')`
I get:
`ValueError: conflicting sizes for dimension 'y': length 128 on the data but length 760 on coordinate 'longitude'`

What am I doing wrong?

Thanks in advance for your support and patience

Jesús de Diego

**Comment by norlandrhagen** [comment]
Hey Jesús,

Glad you found `ndpyramid`. Do you have an MRE (minimal reproducible example) of the dataset you are trying to create pyramids for? That would help us figure out what's going on. 

Also, it might be helpful to run through some of the examples to make sure ndpyramid is working on your machine correctly: https://ndpyramid.readthedocs.io/en/latest/examples/pyramid-resample.html
